test/test_torch.py

```diff
@@ -5404,6 +5404,229 @@ def test_narrow_empty(self, device):
             sz[d] = 0
             self.assertEqual(sz, y.size())
 
+    # This test calls torch.pad and numpy.pad many different ways to ensure
+    # that they give the same results
+    @onlyOnCPUAndCUDA
+    @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False))
+    def test_pad(self, device, dtype):
+        test_cases = [
+            # Input size, pad mode, pad width, kwargs
+            ((10, 10, 10), 'constant', ((0, 0), (0, 0), (0, 0)), {}),
```

Comment:
This is a great way of writing cases and comparing with NumPy, a couple ideas from @saketh-are and me:

- add floating point constant values
- maybe comments could appear every so often to explain which aspect of pad was being tested so it was easier to know all the various input modes were exhaustively tested
- make sure to test each of the 4 modes allowed for the padding to be specified and each of the four modes for specifying values for constant padding (if that's a pain then maybe these could be programmatically generated using a cross-product?)

Reply:
I've added floating and complex values for constant mode, added some cases for each possible native python format, and added comments